src/key.js

```diff
@@ -623,6 +623,86 @@ Key.prototype.revoke = function() {
 };
 
 /**
+ * Signs the public key
+ * @param  {Array<module:key~Key>} privateKey decrypted private keys for signing
+ * @return {module:key~Key} new public key with new certificate signature
+ */
+Key.prototype.sign = function(privateKeys) {
+  var primaryUser, primaryKey, dataToSign, signingKeyPacket, signaturePacket, user, key;
+  if (this.isPrivate()) {
+    throw new Error('Only public key can be signed');
+  }
+  primaryUser = this.getPrimaryUser();
+  if (!primaryUser) {
+    throw new Error('Could not find primary user');
+  }
+  primaryKey = this.primaryKey;
+  dataToSign = {};
+  dataToSign.userid = primaryUser.user.userId || primaryUser.user.userAttribute;
+  dataToSign.key = primaryKey;
```

Comment:
What do u think about for the first time realize functionality in `Key.proto`:
```javascript
Key.proto.signPrimaryUser(privateKeys) => newKey,
Key.proto.signAllUsers(privateKeys) => newKey,
Key.proto.verifyPrimaryUser(publicKeys) => [
    {
        keyid: ...
        valid: ...
    },
    ...
],
Key.proto.verifyAllUsers(publicKeys) => [
    {
        userid: ...
        keyid: ...
        valid: ...
    },
    ...
]
```
And later change inner structure, move some part to User.proto, etc..